### PR TITLE
fix: added missing cronjob LE renewal after import / new install 

### PR DESCRIPTION
**Pull request by NickCoolii**
Fixing #2496 in easy way

### CODE REVIEW

File: bin/v-restore-user [hunk 1]
```diff
@@ -376,6 +381,17 @@ if [ "$web" != 'no' ] && [ -n "$WEB_SYSTEM" ]; then
...
     done
 
+    # if there is at least one domain with LE
+    if $ADD_LETSENCRYPT_CRON_JOB ; then
+        # Adding LE autorenew cronjob
+        if [ -z "$(grep v-update-lets $HESTIA/data/users/admin/cron.conf)" ]; then
+            min=$(generate_password '012345' '2')
+            hour=$(generate_password '1234567' '1')
+            cmd="sudo $BIN/v-update-letsencrypt-ssl"
+            $BIN/v-add-cron-job admin "$min" "$hour" '*' '*' '*' "$cmd" > /dev/null
```
Comment (jaapmarcus) [inline bin/v-restore-user]: There is no need to add this line if we add it to the installers

Similar command In upgrade script will make sure the cronjob runs on existing servers

File: install/hst-install-debian.sh [hunk 2]
```diff
@@ -1977,6 +1977,8 @@ command="sudo $HESTIA/bin/v-update-user-stats"
 $HESTIA/bin/v-add-cron-job 'admin' '20' '00' '*' '*' '*' "$command"
 command="sudo $HESTIA/bin/v-update-sys-rrd"
 $HESTIA/bin/v-add-cron-job 'admin' '*/5' '*' '*' '*' '*' "$command"
+command="sudo $HESTIA/bin/v-update-letsencrypt-ssl"
+$HESTIA/bin/v-add-cron-job 'admin' '30' '07' '*' '*' '*' "$command"
```
Comment (jaapmarcus) [inline install/hst-install-debian.sh]: Random times should be used to limit DDOS effect to LE servers

File: install/hst-install-ubuntu.sh [hunk 3]
```diff
@@ -1997,6 +1997,8 @@ command="sudo $HESTIA/bin/v-update-user-stats"
 $HESTIA/bin/v-add-cron-job 'admin' '20' '00' '*' '*' '*' "$command"
 command="sudo $HESTIA/bin/v-update-sys-rrd"
 $HESTIA/bin/v-add-cron-job 'admin' '*/5' '*' '*' '*' '*' "$command"
+command="sudo $HESTIA/bin/v-update-letsencrypt-ssl"
+$HESTIA/bin/v-add-cron-job 'admin' '30' '07' '*' '*' '*' "$command"
```
Comment (jaapmarcus) [inline install/hst-install-ubuntu.sh]: Random times should be used to limit DDOS effect to LE servers

**Review by jaapmarcus — CHANGES_REQUESTED** [review]

File: bin/v-restore-user (hunk 1 above)
Comment (NickCoolii) [reply bin/v-restore-user:391]: But there are can be existing installations without LE cronjob which will get update and after restore of user there will be no cronjob for LE.
So this is patching existing installations.
Is there some way to add cronjob on update?
But I agree with you that this code should be marked as temporary and removed in future.

Comment (NickCoolii) [reply bin/v-restore-user]: I can remove it from here, but I don't know where it should be added for upgrade.

File: install/hst-install-debian.sh (hunk 2 above)
Comment (NickCoolii) [reply install/hst-install-debian.sh]: Function `generate_password()` defined at `func/main.sh`
https://github.com/hestiacp/hestiacp/blob/a26eea3721d720f7a1719081578594033f1d7f64/func/main.sh#L213-L224

So should I duplicate this func at `hst-install-*.sh` if we need it to be used?
There are `gen_pass()` at `hst-install-*.sh` but it looks like first version of  `generate_password()` and didn't get arguments.
https://github.com/hestiacp/hestiacp/blob/9fef8b0626db078db2a5c1b30dd7bfc7318b91bf/install/hst-install-debian.sh#L118-L121
So we can add new one for randomising min and hour for cronjob or modify this one.
Which way will be better?

Comment (jaapmarcus) [reply install/hst-install-debian.sh]: Modifiing should be fine

File: bin/v-restore-user (hunk 1 above)
Comment (jaapmarcus) [reply bin/v-restore-user]: Create a 1.6.0.sh in /install/upgrade/versions/

And use as a template

https://github.com/hestiacp/hestiacp/blob/main/install/upgrade/versions/1.5.9.sh

👍🏼

File: install/upgrade/versions/1.6.0.sh [hunk 4]
```diff
@@ -20,3 +20,15 @@ upgrade_config_set_value 'UPGRADE_UPDATE_DNS_TEMPLATES' 'false'
 upgrade_config_set_value 'UPGRADE_UPDATE_MAIL_TEMPLATES' 'false'
 upgrade_config_set_value 'UPGRADE_REBUILD_USERS' 'false'
 upgrade_config_set_value 'UPGRADE_UPDATE_FILEMANAGER_CONFIG' 'false'
+
+# Includes
+# shellcheck source=/usr/local/hestia/func/main.sh
+source $HESTIA/func/main.sh
```
Comment (jaapmarcus) [inline install/upgrade/versions/1.6.0.sh]: Is not need already loaded at the start of the upgrade procedure

**Review by jaapmarcus — APPROVED** [review]

**Pull request merged**